src/Quaternion.jl

```diff
@@ -29,14 +39,72 @@ quat(p, v1, v2, v3, n) = Quaternion(p, v1, v2, v3, n)
 quat(x) = Quaternion(x)
 quat(s, a) = Quaternion(s, a)
 
+"""
+    real(T::Type)
```

Comment:
This is too general. Let's do specifically
```suggestion
    real(T::Type{<:Quaternion})
```